gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -64,11 +71,13 @@
 public class BatcherImpl<ElementT, ElementResultT, RequestT, ResponseT>
     implements Batcher<ElementT, ElementResultT> {
 
+  private static final Logger LOG = Logger.getLogger(BatcherImpl.class.getName());
   private final BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT>
       batchingDescriptor;
   private final UnaryCallable<RequestT, ResponseT> unaryCallable;
   private final RequestT prototype;
   private final BatchingSettings batchingSettings;
+  private final BatcherReference phantom;
```

Comment:
It looks like most of this PR mimics the grpc's implementation of [ManagedChannelReference](https://github.com/grpc/grpc-java/blob/48ca4527c14a95914f9cb7f58ec72997cb96899a/core/src/main/java/io/grpc/internal/ManagedChannelImpl.java#L1262), so that is I'm not questioning the stylistic things in this PR (some of them are questionable, but for sake of consistency we can keep them). 

But one small thing anyways. This variable is called `phantom` (same in grpc, so "consistent"). At the same time, the actual type of the referrence is [WeakReferrence](https://docs.oracle.com/javase/7/docs/api/java/lang/ref/WeakReference.html). Since JDK has [PhantomReference](https://docs.oracle.com/javase/7/docs/api/java/lang/ref/PhantomReference.html) as well, I think calling this field "phantom" is incorrect (maybe historically in grpc implementation it used to be `PhantomReference`, and then was changed to `WearkReference` without renaming the actual variable). Please rename it to something, which does not make wrong impression on it (that it is a phantom referrence instead of weak).

Reply:
Thanks for the feedback. I updated the BatcherReference variable name to `currentBatcherReference'.